_posts/2023-04-30-openai-supabase-summary.md

```diff
@@ -0,0 +1,175 @@
+                + "\n"
+                + "Here is an example output:"
+                + "\n"
+                + params.prompt_example
+                + "\n"
+                + "Here is the data: "
+                + str(documents)
+                + "\n"
+                + "And here is the previous analysis: "
+                + last_analysis,
+            },
+        ],
+        temperature=0,
+        max_tokens=256,
+        top_p=1.0,
+        frequency_penalty=0.0,
+        presence_penalty=0.0,
+    )
+    return response.choices[0].message.content
+```
+
+The `gpt_4_summarizer` step is a custom ZenML step that leverages GPT-4 to generate summaries of the input data. It takes `SummarizerParams` and a list of documents as input, along with the `StepContext` to access previous pipeline runs.
+
+The step first retrieves the secret API key for OpenAI and checks for any previous analysis. If a previous analysis is found, it is included in the input to GPT-4. This allows GPT-4 to consider the previous analysis while generating the new summary.
+
+The step then calls the GPT-4 API with the specified parameters and input data, including the documents and the previous analysis. The API returns a summary, which is then returned as the output of the step.
+
+This custom step can be adapted for other use cases by modifying the input parameters, the GPT-4 model, or the API call. For example, you could use this step to generate summaries for different types of data, use a different language model, or customize the API call to suit your specific needs. By customizing this step, you can harness the power of GPT-4 and Zen
+
+### 📆 Daily reports on Slack
+
+<figure>
+  <img src="/assets/posts/gpt-4-supabase-summary/youtldr_summarizer_slack.png" alt="Analysis is reported to Slack every day.">
+  <figcaption style="text-align:center;font-size:14px;color:#666;">Analysis is reported to Slack every day.</figcaption>
+</figure>
+
+## 🚀 Overcoming GPT-4 Limitations with ZenML
+
+One of the key insights of this case study is that by versioning the summaries over multiple pipelines, we can overcome GPT-4's context limitations. Instead of comparing raw data, we can compare summaries generated by GPT-4, allowing for more meaningful insights and analysis.
+
+```mermaid
+graph TB
+A[you-tldr.com] --> B[Supabase Analytics Table]
+B --> C[ZenML Pipeline]
+C --> D[GPT-4 Analysis]
+D --> E[Versioned Summaries]
+E --> F[GPT-4 Comparison]
+F --> G[Insights]
+```
+
+**Figure 4:** System Architecture Diagram
+
+## 🎯 Conclusion
+
+This case study demonstrates the power of combining ZenML with GPT-4 to analyze and version data from a Supabase database. By using ZenML to manage data versioning and GPT-4 for analysis, we can overcome the limitations of LLMs and gain valuable insights from our data. If you're interested in leveraging the latest technology for your own projects, consider using ZenML in conjunction with GPT-4 and Supabase to unlock the full potential of your data.
```

Comment:
Probably wants some links or something back to ZenML, maybe to other LLM- or OpenAI- related posts and docs etc? Also to the project repo itself?